### ISSUE TITLE
KMS keyring generator key id

### ISSUE THREAD

**Issue by mattsb42-aws** [aws/aws-encryption-sdk-c#373]
If I am reading this correctly, the [KMS keyring builder](https://github.com/aws/aws-encryption-sdk-c/blob/master/aws-encryption-sdk-cpp/include/aws/cryptosdk/cpp/kms_keyring.h#L128-L129) requires that `generator_key_id` is set. This should not be required.

**Comment by david-koenig** [comment]
The non-API-breaking way to make this change is to allow NULL to be passed in for the generator key ID. We could additionally allow an empty string to be passed in and treat it the same way.

**Comment by mattsb42-aws** [comment]
@david-koenig Agreed. This should have a similar pattern as creating a multikeyring with no generator.

**Comment by robin-aws** [comment]
See also #557
